app/web/src/workers/webworker.ts

```diff
@@ -1432,6 +1432,25 @@ const pruneAtomsForClosedChangeSet = async (
   });
 };
 
+const atomExistsWithChecksum = (
+  kind: EntityKind,
+  id: string,
+  checksum: string,
+) => {
+  const result = [];
+  const rows = db.exec({
+    sql: `select * from atoms where atoms.kind = ? AND atoms.args = ? and atoms.checksum = ?;`,
+    bind: [kind, id, checksum],
+    returnValue: "resultRows",
+  });
+
+  for (const row of rows) {
+    result.push(row[0]);
+  }
+
+  return result.length > 0;
```

Comment:
Why the manual row count, instead of doing `select count(*) ...` and have the query return it directly?